packages/react/src/components/Storage/FileUploader/hooks/useFileUploader/types.ts

```diff
@@ -0,0 +1,20 @@
+import { FileStatuses } from '../../types';
+
+export interface DragActionHandlers {
+  onDragStart: (event: React.DragEvent<HTMLDivElement>) => void;
+  onDragEnter: (event: React.DragEvent<HTMLDivElement>) => void;
+  onDragLeave: (event: React.DragEvent<HTMLDivElement>) => void;
+  onDrop: (event: React.DragEvent<HTMLDivElement>) => void;
+  onDragOver: (event: React.DragEvent<HTMLDivElement>) => void;
+}
+
+export interface UseFileUploader extends DragActionHandlers {
+  autoUploadRef: React.MutableRefObject<boolean>;
```

Comment:
I'll need some advice on this one. This is tricky since this is used to check if we need to proceed to automatically uploading after a file is added if `autoProceed` is turned on. If it is, this `useEffect` triggers the function to start the upload.

```
useEffect(() => {
    if (shouldAutoProceed && autoUploadRef.current) {
      onFileClick();
    }
    autoUploadRef.current = false;
  }, [shouldAutoProceed, autoUploadRef, onFileClick]);
```

The `autoUploadRef` is passed into the `useFileUploader` hook since it needs to trigger when a user drops a file. This is also trigged in the `FileUploader` in `onFileChange` when a user presses a button and adds a file.

For the button press, it's easy to trigger the upload (since the code is both in the `FileUploader` component), however the drop code is in the `useFileUploader` hook. So I can't easily pass in the `onFileClick` to it, to start the upload, after a user drops a file. I could try moving the `onFileClick` code all together, but that's really complicated and it's a big refactor.

Any suggestions would be helpful @calebpollman  and @dbanksdesign . Or maybe we can leave as is?

Reply:
I think I found a solution, I moved it from a ref to a setState and that seemed to work. I'll leave it as is.

Reply:
[9c0abe1](https://github.com/aws-amplify/amplify-ui/pull/3028/commits/9c0abe1eff6686f581303855a9f44ebf63a6ccfc)